packages/suite/src/utils/wallet/coinControlUtils.ts

```diff
@@ -0,0 +1,31 @@
+import { AccountOutputLabels } from '@suite-common/metadata-types';
+import { Utxo } from '@trezor/blockchain-link';
+
+/**
+ * Filter UTXOs based on search query and categorize them into spendable, low anonymity and dust UTXOs.
+ */
+export const filterAndCategorizeUtxos = (
+    searchQuery: string,
+    utxos: Utxo[],
+    spendableUtxos: Utxo[],
+    lowAnonymityUtxos: Utxo[],
+    dustUtxos: Utxo[],
+    outputLabels: { [txid: string]: AccountOutputLabels }
+) => {
+    const filteredUtxos = utxos.filter(utxo =>
+        utxo.address.includes(searchQuery) ||
+        utxo.txid.includes(searchQuery) ||
+        outputLabels?.[utxo.txid]?.[utxo.vout].includes(searchQuery)
+    );
+
+    const filteredSpendableUtxos = filteredUtxos.filter(utxo => spendableUtxos.includes(utxo));
+    const filteredLowAnonymityUtxos = filteredUtxos.filter(utxo => lowAnonymityUtxos.includes(utxo));
+    const filteredDustUtxos = filteredUtxos.filter(utxo => dustUtxos.includes(utxo));
```

Comment:
This can be written (I believe) like this: 

```ts
const filterFunction = (utxo: Utxo) =>
        utxo.address.includes(searchQuery) ||
        utxo.txid.includes(searchQuery) ||
        outputLabels?.[utxo.txid]?.[utxo.vout].includes(searchQuery);

    return {
        filteredUtxos: utxos.filter(filterFunction),
        filteredSpendableUtxos: spendableUtxos.filter(filterFunction),
        filteredLowAnonymityUtxos: lowAnonymityUtxos.filter(filterFunction),
        filteredDustUtxos: dustUtxos.filter(filterFunction),
    };
    ```